### PR TITLE
BmcITSM- fix duplicate incidents

### PR DESCRIPTION
**Pull request by maimorag**
## Status
- [x] In Progress

## Related Issues
fixes:[ link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-38910)

## Description
Adding logs

### CODE REVIEW

**Comment by YuvHayun** [comment]
logs looks good.

File: Packs/BmcITSM/ReleaseNotes/1_0_23.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### BMC Helix ITSM
+
+- Fixed an issue where ***fetch-incidents*** fetched duplicate incidents.
```
Comment (YuvHayun) [inline Packs/BmcITSM/ReleaseNotes/1_0_23.md:6]: due to?
Always good to leave a comment why for future developers.

**Review by YuvHayun — APPROVED** [review]
Looks good!
Minor comment.

File: Packs/BmcITSM/ReleaseNotes/1_0_23.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+#### Integrations
+
+##### BMC Helix ITSM
+
+- Fixed an issue where ***fetch-incidents*** fetched duplicate incidents because the integration incorrectly assumed that fetched incidents are always sorted in ascending order and set the last fetched time based on the last incident time.
```
Comment (dantavori) [inline Packs/BmcITSM/ReleaseNotes/1_0_23.md:6]:
```suggestion
- Fixed an issue in the ***fetch-incidents*** command where duplicate incidents were fetched due to the incorrect assumption that tickets pulled from BMC Helix ITSM are sorted in ascending order.
```

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py [hunk 3]
```diff
@@ -3340,7 +3339,9 @@ def fetch_relevant_tickets(
         tickets_capacity -= tickets_amount
 
         if fetched_tickets:
-            last_ticket_create_time = total_tickets[-1].get("CreateDate")
+            last_ticket_create_time = max(
+                [date_to_epoch_for_fetch(arg_to_datetime(ticket.get("CreateDate")))
+                 for ticket in total_tickets])
```
Comment (dantavori) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py:3344]: won't we run now date_to_epoch_for_fetch twice?

Comment (maimorag) [reply Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py:3344]: yes, thanks will fix it

File: Packs/BmcITSM/.pack-ignore [hunk 4]
```diff
@@ -1,2 +1,11 @@
...
+
 [file:classifier-BMC_ITSM_Outgoing_Mapper.json]
-ignore=MP106
\ No newline at end of file
+ignore=MP106
+
+
+XSOAR build: dev-v8.8.0-1146902-1ba0b574 (branch: dev)
+BE3 build: dev-v3.12.0-2446-ge4716 (branch: dev)
+FE build: xsiam--dev-v3.12.0-2360-gbacbeb9d37 (branch: dev)
```
Comment (dantavori) [inline Packs/BmcITSM/.pack-ignore:11]: why is this needed?

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM_test.py [hunk 5]
```diff
@@ -1480,3 +1482,38 @@ def test_ticket_list_work_order_command(
...
+                                                                impact_filter=[],
+                                                                urgency_filter=[],
+                                                                custom_query=("'Submit Date' <= \"1657032797\" AND 'Submit Date'"
+                                                                              ">\"1657032797\" AND 'Urgency' = \"4-Low\""),
+                                                                mirror_direction="both",
+                                                                )
+    expected_result = max(
+        [BmcITSM.date_to_epoch_for_fetch(arg_to_datetime(ticket.get("occured")))
+         for ticket in incidents_result]
+    )
```
Comment (dantavori) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM_test.py:1518]: in a test you are not supposed to re-implement the logic of what you test - better to just write the expected result, hard-coded

**Review by dantavori — APPROVED** [review]
approved, but see notes
